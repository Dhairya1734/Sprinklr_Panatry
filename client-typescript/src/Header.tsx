import React from "react";

const Header: React.FunctionComponent = () => {
	// eslint-disable-next-line no-console
	// console.log("Header");
	return <header id='heading' data-testid="siteHeader">Sprinklr Pantry</header>;
};

export default React.memo(Header);
