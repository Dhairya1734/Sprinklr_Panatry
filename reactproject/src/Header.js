import React from 'react';

export default React.memo(function Header(){
    console.log("Header")
    return(
    <header id="heading">
        Sprinklr Pantry
    </header>
    );
});