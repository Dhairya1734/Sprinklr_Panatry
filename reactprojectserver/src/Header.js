import React from 'react';
import PropTypes from 'prop-types';

function returnList(num){
    let allList=[];
    for(let i=1;i<=num;i++){
        allList.push(<option key={i} value={i<10 ? "0" + i.toString() : i.toString()}>{i}</option>);
    }
    return allList;
}

function Header(props){

    return (
        <header id="heading">
            <span id="headingName">Sprinklr Pantry</span>
            <section className="search">
                <label> Table Number: 
                    <select className="findTable" onChange ={props.handler} name="findTable" > 
                        <option key="all" value="all"> All</option>
                        {returnList(10)}
                    </select>
                </label>
                    {/* {<button className="searchButton" onClick = {(e) => console.log(document.querySelector('.Find_Table').value)}> OK </button>} */}
            </section>
        </header>

    );

};

Header.propTypes = {
    handler : PropTypes.func.isRequired,
}

export default React.memo(Header);