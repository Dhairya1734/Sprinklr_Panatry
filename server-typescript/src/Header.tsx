import React from 'react';
import PropTypes from 'prop-types';

type Props = {
    onSetTable : (e: React.ChangeEvent<HTMLSelectElement>) => void,
    searchRef : React.RefObject<HTMLSelectElement>;
}

function returnList(num : number) : JSX.Element[] {
    let allList : JSX.Element[]=[];
    for(let i=1;i<=num;i++){
        allList.push(<option key={i} value={i<10 ? "0" + i.toString() : i.toString()}>{i}</option>);
    }
    return allList;
}

const Header = (props : Props) : JSX.Element => {

    return (
        <header id="heading">
            <span id="headingName">Sprinklr Pantry</span>
            <section className="search">
                <label> Table Number: 
                    <select className="findTable" onChange ={props.onSetTable} name="findTable" ref = {props.searchRef}> 
                        <option key="all" value="all"> All</option>
                        {returnList(10)}
                    </select>
                </label>
                    {/* {<button className="searchButton" onClick = {(e) => console.log(document.querySelector('.Find_Table').value)}> OK </button>} */}
            </section>
        </header>

    );

};

Header.propTypes = {
    onSetTable : PropTypes.func.isRequired,
    searchRef : PropTypes.object,
}

export default React.memo(Header);
