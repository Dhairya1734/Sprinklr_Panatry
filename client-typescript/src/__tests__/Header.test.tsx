// import * as React from "react";
import { render  } from "@testing-library/react";
import Header from "../Header";
import "@testing-library/jest-dom";

test("Header Test", () => {
	const { getByTestId } = render(<Header />);
	const element = getByTestId("siteHeader");
	expect(element).toHaveTextContent("Sprinklr Pantry");
	expect(element).toMatchSnapshot();
});
