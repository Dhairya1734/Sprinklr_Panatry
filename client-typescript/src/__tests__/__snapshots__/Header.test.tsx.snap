// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header Test 1`] = `
<header
  data-testid="siteHeader"
  id="heading"
>
  Sprinklr Pantry
</header>
`;
